text/0028-bill-of-materials.md

```diff
@@ -0,0 +1,95 @@
+
+## Summary
+
+A bill of materials (BOM) is a set of metadata that provides an account of the
+dependencies and tools that are both present in an application image, as well
+those used in the construction of that image. The availability of this data is
+a key value proposition of Cloud Native Buildpacks.
+
+As an implementation of Cloud Native Buildpacks, the Paketo project should
+support adding bill of materials metadata across the different components that
+go into an app image.
+
+## Motivation
+
+A BOM can include an array of different metadata about application dependencies
+and buildpack dependencies such as SHAs, URIs, versions, license information,
+vulnerability identifiers, etc. This information is valuable for users who want
+an accurate inventory of everything in their final application image, or that
+may have been used to build it. Having access to that information makes it
+easier to audit for vulnerabilities and license information.
+
+The Paketo Java Buildpacks as well as a handful of other Paketo Buildpacks
+already make the BOM available. Another motivation for doing this work is to
+achieve a consistent user experience across the Paketo project.
+
+## Detailed Explanation
+
+The following Paketo components should have BOM metadata attached to them:
+
+### Stacks
+Stacks (such as those found in the [Paketo Stacks repository](https://github.com/paketo-buildpacks/stacks))
+should have BOM metadata that includes in-depth information on all of the OS
+level packages installed as part of the stack.
+
+The existing metadata for stacks has the following structure:
+```
+{
+  "name" : "<package name>",
+  "version" : "<package version>",
+  "arch" : "<compatible architecture>",
+  "source" : {
+    "name" : "<package source name>",
+    "version" : "<package source version>",
+    "upstreamVersion" : "<package source upstream version>"
+  },
+  "summary" : "<package summary>"
+}
+```
+
+### Runtime and Compilation Dependencies
+Dependencies that provide runtimes and/or are used for compilation should have
+BOM metadata surfaced about them. This includes both the dependencies in the
+final application image, as well as those used during the image building
+process. An example of this type of dependency is the node-engine dependency
+that is provided by the [Paketo Node Engine Buildpack](https://github.com/paketo-buildpacks/node-engine).
+
+The standardized set of keys to include in dependency BOM entries are:
+```
+[[bom]]
+name = "<dependency name>"
+
+[bom.metadata]
+  sha256 = "<hash of dependency artifact from uri>"
+  uri = "<uri to dependency>"
+  version = "<dependency version>"
+
+  # Optional parameters
+  stacks = [<list of compatible stacks>]
+  cpe = "<version-specific common platfrom enumeration>"
+  licenses = [<licenses that the dependency has>]
+  source-uri = "<uri to the dependency source>"
+  source-sha256 = "<hash of the dependency source artifact from source-uri>"
+```
+
+### Language Specific Modules
```

Comment:
Calling this "Language Specific Modules" might be too restrictive. You could have non-language specific tools that install additional dependencies. For example, there are tools like `mvnw` and `gradlew` that are wrapper scripts to install Maven & Gradle, and Rust has `rustup` which is a tool for installing Rust.

I see two kinds of dependencies that are installed: 1.) Directly install dependencies and 2.) Indirectly installed dependencies. The former would be directly installed by the buildpack like runtimes & compilation tools (things that are listed in buildpack.toml). The latter would be things like NPM or Maven installed modules, but could be anything that a tool run by the buildpack installs like `rustup` and the Maven/Gradle wrappers.

I think it might help for clarity to have the title of this section and the `Runtime and Compilation Dependencies` section be more generic, and have the present titles be listed as specific examples under each section.

Reply:
Great idea, thanks for the feedback. I tried to address this comment in https://github.com/paketo-buildpacks/rfcs/pull/57/commits/59e94453a4f0878a26b708553b532e3713324902